### ISSUE TITLE
`LNK1104: cannot open file 'python3.lib'` in Windows Python 3.13 RC

### ISSUE THREAD

**Issue by dofuuz** [scikit-build/scikit-build-core#864]
Build fails when targeting ABI3 with Windows + Python 3.13.

OS: Windows
Python: 3.13.0rc1

```
   LINK : fatal error LNK1104: cannot open file 'python3.lib' [D:\a\nanobind_example\nanobind_example\build\cp312-abi3-win_amd64\nanobind_example_ext.vcxproj]
```
Full log: https://github.com/dofuuz/nanobind_example/actions/runs/10452840733/job/28942106258#step:5:96
Source: https://github.com/dofuuz/nanobind_example/tree/43c312cfa2dba30183c66a1f8b5ed31c8e0b8083

python3.lib exists within Python libs folder.

More simple way to reproduce it from scikit-build-core example:
```
cd scikit-build-core\tests\packages\abi3_pyproject_ext
pip wheel -v .
```

**Comment by LecrisUT** [comment]
Can you check if `setup-python` provides full dev packages in this case? Otherwise it could be some path escaping is not done properly in nanobind :shrug:

**Comment by dofuuz** [comment]
@LecrisUT

I could reproduce it from local Windows PC, so it's not problem of `setup-python`.

And same error occurs in abi3_pyproject_ext example of scikit-build-core. So it's not problem of nanobind.

**Comment by LecrisUT** [comment]
Still it is most probably a CMake issue, but I couldn't find any reports in the [upstream issues](https://gitlab.kitware.com/cmake/cmake/-/issues. Managed to capture the test failure in `tests\test_pyproject_abi3.py`. I wonder if the issue is with `Development.SABIModule` because it seems the tests that don't use it did not fail :thinking:

**Comment by henryiii** [comment]
Even if it is, we likely can fix it via setting `Python_SABI_LIBRARY`.

**Comment by henryiii** [comment]
Debugging in #865

**Comment by henryiii** [comment]
I have a feeling we are being affected by https://github.com/python/cpython/pull/110049/files#diff-d593bd299ba58e440ba411ffa0640ccd9d20d518b0cf2644ed4bdb75a82a3e70 combined with some logic issues, some of these variables were missing on Windows before.

**Comment by henryiii** [comment]
Looks like that fixes our tests, at least.

**Comment by henryiii** [comment]
Should be fixed in 0.10.4!

**Comment by dofuuz** [comment]
Fix confirmed! Thank you 👍
